apis/core/v1beta1/openstackcontrolplane_webhook.go

```diff
@@ -150,7 +150,7 @@ func (r *OpenStackControlPlane) checkDepsEnabled(name string) string {
 }
 
 // ValidateServices implements common function for validating services
-func (r *OpenStackControlPlane) ValidateServices(basePath *field.Path) field.ErrorList {
+func (r *OpenStackControlPlane) ValidateServices(basePath *field.Path, isUpdate bool) field.ErrorList {
```

Comment:
Booleans in function parameters that then used in ifs to change logic is a common code smell. This function probably need to be two separate functions, one for the create and one for the update codepath.